app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt

```diff
@@ -389,11 +388,17 @@ class ProjectActivity : BaseActivity<ProjectViewModel.ViewModel>() {
         return Pair.create(R.anim.fade_in_slide_in_left, R.anim.slide_out_right)
     }
 
-    override fun back() {
+    override fun back() = if (this.nativeCheckout) {
+        val rewardsSheetIsExpanded = pledge_container.alpha == 1f
         when {
-            supportFragmentManager.backStackEntryCount > 0 -> supportFragmentManager.popBackStack()
-            action_buttons.visibility == View.GONE -> this.viewModel.inputs.hideRewardsFragmentClicked()
-            else -> super.back()
+            supportFragmentManager.backStackEntryCount > 0 && rewardsSheetIsExpanded -> supportFragmentManager.popBackStack()
+            rewardsSheetIsExpanded -> this.viewModel.inputs.hideRewardsFragmentClicked()
+            else -> {
+                supportFragmentManager.popBackStack()
```

Comment:
This feels fickle if we ever have multiple fragments on the stack. I think you want to use `.popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE);`